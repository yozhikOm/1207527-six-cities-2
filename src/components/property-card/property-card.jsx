import React from 'react';
import PropTypes from 'prop-types';

const PropertyCard = ({offerInfo, cardMouseEnterHandler}) => {
  const {id, title, type, price, isPremium, photos} = offerInfo;

  const onCardMouseEnter = (evt) => {
    const cardId = Number(evt.currentTarget.id);
    cardMouseEnterHandler(cardId);
  };

  const onCardTitleClick = (evt) => {
    console.log(`'` + evt.target.textContent + `' has just been clicked`);
    console.log(id + `: '` + title + `' has just been clicked`);
    window.location.assign(`/offer/` + id);
  };

  return (
    <article className="cities__place-card place-card" id={id} onMouseEnter={onCardMouseEnter}>
      {isPremium ?
        <div className="place-card__mark">
          <span>Premium</span>
        </div> :
        <React.Fragment/>
      }
      <div className="cities__image-wrapper place-card__image-wrapper">
        <a href="#">
          <img className="place-card__image" src={photos[0].src} width="260" height="200" alt="Place image" />
        </a>
      </div>
      <div className="place-card__info">
        <div className="place-card__price-wrapper">
          <div className="place-card__price">
            <b className="place-card__price-value">&euro;{price}</b>
            <span className="place-card__price-text">&#47;&nbsp;night</span>
          </div>
          <button className="place-card__bookmark-button button" type="button">
            <span className="visually-hidden">To bookmarks</span>
          </button>
        </div>
        <div className="place-card__rating rating">
          <div className="place-card__stars rating__stars">
            <span className="visually-hidden">Rating</span>
          </div>
        </div>
        <h2 className="place-card__name" onClick={onCardTitleClick}>
          <a href="#">{title}</a>
        </h2>
        <p className="place-card__type">{type}</p>
      </div>
    </article>
  );
};

PropertyCard.propTypes = {
  offerInfo: PropTypes.shape({
    id: PropTypes.number.isRequired,
    location: PropTypes.shape({
      city: PropTypes.string,
      coordinates: PropTypes.arrayOf(PropTypes.number, PropTypes.number).isRequired,
    }).isRequired,
    title: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    isPremium: PropTypes.bool,
    description: PropTypes.string.isRequired,
    photos: PropTypes.arrayOf(
        PropTypes.shape({
          src: PropTypes.string,
        })
    ),
    host: PropTypes.string.isRequired,
  }),
  cardMouseEnterHandler: PropTypes.func,
};

PropertyCard.defaultProps = {
  id: 1,
  title: `Some title`,
  type: `Room`,
  price: 88,
  description: ``,
  photos: [],
  host: ``
};

export {PropertyCard};
