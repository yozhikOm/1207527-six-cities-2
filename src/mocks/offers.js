export const offers = [
  {"city": {
    "name": `Cologne`,
    "location": {"latitude": 50.938361, "longitude": 6.959974, "zoom": 13}},
  "previewImage": `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
  "images": [`https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`],
  "title": `Penthouse, 4-5 rooms + 5 balconies`,
  "isFavorite": false,
  "isPremium": false,
  "rating": 3.6,
  "type": `apartment`,
  "bedrooms": 3,
  "maxAdults": 5,
  "price": 459,
  "goods": [`Baby seat`, `Fridge`, `Towels`],
  "host": {
    "id": 25,
    "name": `Angelina`,
    "isPro": true,
    "avatarUrl": `img/avatar-angelina.jpg`
  },
  "description": `I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.`,
  "location": {"latitude": 50.957361, "longitude": 6.9509739999999995, "zoom": 16},
  "id": 1
  },
  {"city": {"name": `Brussels`, "location": {"latitude": 50.846557, "longitude": 4.351697, "zoom": 13}}, "previewImage": `https://es31-server.appspot.com/six-cities/static/hotel/13.jpg`, "images": [`https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/16.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/10.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/13.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/8.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/3.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`], "title": `Waterfront with extraordinary view`, "isFavorite": false, "isPremium": false, "rating": 2.5, "type": `house`, "bedrooms": 5, "maxAdults": 10, "price": 474, "goods": [`Laptop friendly workspace`], "host": {"id": 25, "name": `Angelina`, "isPro": true, "avatarUrl": `img/avatar-angelina.jpg`}, "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`, "location": {"latitude": 50.869557, "longitude": 4.332697, "zoom": 16}, "id": 2},
  {"city": {"name": `Dusseldorf`, "location": {"latitude": 51.225402, "longitude": 6.776314, "zoom": 13}}, "previewImage": `https://es31-server.appspot.com/six-cities/static/hotel/16.jpg`, "images": [`https://es31-server.appspot.com/six-cities/static/hotel/10.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/13.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/8.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/7.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/9.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/11.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`], "title": `Waterfront with extraordinary view`, "isFavorite": false, "isPremium": false, "rating": 2.4, "type": `hotel`, "bedrooms": 3, "maxAdults": 5, "price": 474, "goods": [`Towels`, `Laptop friendly workspace`, `Baby seat`, `Washer`, `Air conditioning`, `Breakfast`], "host": {"id": 25, "name": `Angelina`, "isPro": true, "avatarUrl": `img/avatar-angelina.jpg`}, "description": `Design interior in most sympathetic area! Complitely renovated, well-equipped, cosy studio in idyllic, over 100 years old wooden house. Calm street, fast connection to center and airport.`, "location": {"latitude": 51.216402, "longitude": 6.758314, "zoom": 16}, "id": 3}
];
