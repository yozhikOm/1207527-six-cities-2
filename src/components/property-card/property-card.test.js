import React from 'react';
import {PropertyCard} from './property-card.jsx';
import renderer from 'react-test-renderer';
import {offers} from '../../mocks/offers.js';

it(`PropertyCard компонент рендерится корректно`, () => {
  const propertyCardComponent = renderer.create(<PropertyCard offerInfo={offers[0]}/>).toJSON();
  expect(propertyCardComponent).toMatchSnapshot();
});
