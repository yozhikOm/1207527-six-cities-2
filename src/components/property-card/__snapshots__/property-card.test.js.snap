// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard компонент рендерится корректно 1`] = `
<article
  className="cities__place-card place-card"
  id={1}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="https://es31-server.appspot.com/six-cities/static/hotel/2.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          459
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button"
        type="button"
      >
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "93%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
      onClick={[Function]}
    >
      <a
        href="#"
      >
        Penthouse, 4-5 rooms + 5 balconies
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      apartment
    </p>
  </div>
</article>
`;
